### PR TITLE
Changed the way Job ID is passed down to internal scheduler commands …

### PR DESCRIPTION
**Pull request by amrhassan**
…to fix communication protocol errors

@muhamadazmy please review and merge if all is okay.

### CODE REVIEW

File: lib/JumpScale/lib/agentcontroller2client/acclient.py [hunk 1]
```diff
@@ -621,7 +624,7 @@ def schedule_remove(self, id, validate_queued=False):
         """
         Remove a scheduled job by ID
         """
-        cmd = self.cmd(0, 0, 'controller', RunArgs(name='scheduler_remove'), id=id, roles=['*'])
+        cmd = self.cmd(0, 0, 'controller', RunArgs(name='scheduler_remove'), data=str(id), roles=['*'])
```
Comment (muhamadazmy) [inline lib/JumpScale/lib/agentcontroller2client/acclient.py]: Use `data=json.dumps(id)` instead. and change AC2 side to load json of data isntead.

File: lib/JumpScale/lib/agentcontroller2client/acclient.py [hunk 2]
```diff
@@ -630,7 +633,7 @@ def schedule_remove_prefix(self, prefix, validate_queued=False):
         """
         Remove a scheduled job by ID
         """
-        cmd = self.cmd(0, 0, 'controller', RunArgs(name='scheduler_remove_prefix'), id=prefix, roles=['*'])
+        cmd = self.cmd(0, 0, 'controller', RunArgs(name='scheduler_remove_prefix'), data=str(id), roles=['*'])
```
Comment (muhamadazmy) [inline lib/JumpScale/lib/agentcontroller2client/acclient.py]: Same as previous comment

File: lib/JumpScale/lib/agentcontroller2client/acclient.py [hunk 3]
```diff
@@ -557,7 +557,9 @@ def cmd(self, gid, nid, cmd, args, data=None, id=None, roles=None, fanout=False,
         cmd = self._build_cmd(gid=gid, nid=nid, cmd=cmd, args=args, data=data,
                               id=id, roles=roles, fanout=fanout, tags=tags)
 
+
         payload = json.dumps(cmd.dump())
+	print('Executing command: %s' % cmd)
```
Comment (muhamadazmy) [inline lib/JumpScale/lib/agentcontroller2client/acclient.py]: Remove this print statement.

File: lib/JumpScale/lib/agentcontroller2client/acclient.py (hunk 1 above)
Comment (amrhassan) [inline lib/JumpScale/lib/agentcontroller2client/acclient.py]: Changed to `data=json.dumps(str(id))` and made sure AgentController2 always expects a JSON string.

File: lib/JumpScale/lib/agentcontroller2client/acclient.py (hunk 2 above)
Comment (amrhassan) [inline lib/JumpScale/lib/agentcontroller2client/acclient.py]: Changed as well.

File: lib/JumpScale/lib/agentcontroller2client/acclient.py (hunk 3 above)
Comment (amrhassan) [inline lib/JumpScale/lib/agentcontroller2client/acclient.py]: Removed

**Pull request merged**